core/src/test/java/org/elasticsearch/action/admin/indices/template/put/PutIndexTemplateRequestTests.java

```diff
@@ -107,4 +111,21 @@ public void testPutIndexTemplateRequestSerializationXContentBwc() throws IOExcep
             assertEquals("template", request.patterns().get(0));
         }
     }
+
+    public void testValidateErrorMessage() throws Exception {
+        PutIndexTemplateRequest request = new PutIndexTemplateRequest();
+        ActionRequestValidationException withoutNameAndPattern = request.validate();
+        assertThat(withoutNameAndPattern.getMessage(), containsString("name is missing"));
+        assertThat(withoutNameAndPattern.getMessage(), containsString("index patterns are missing"));
+
+        request.name("foo");
+        ActionRequestValidationException withoutIndexPatterns = request.validate();
+        assertThat(withoutIndexPatterns.validationErrors(), hasSize(1));
+        assertThat(withoutIndexPatterns.getMessage(), containsString("index patterns are missing"));
+
+        request.patterns(Collections.singletonList("test-*"));
+        ActionRequestValidationException noError = request.validate();
+        assertTrue(noError == null || noError.validationErrors().isEmpty());
```

Comment:
I don't like `assertTrue`, it never gives useful failure messages (just expected true but was false). Also, I'm missing why this might be null or have empty validation errors. Can you explain why this is not definite?

Reply:
I did not read the function carefully. This is addressed, could you please have another review?